### PR TITLE
load large sample file from bucket

### PR DESCRIPTION
**Pull request by LukeLoze1**
To load a 3.5 Million row file from a gcp bucket
It takes 4.5 hrs in the perf test environment

The test is marked @skip and is ignored with tag processing

https://trello.com/c/zSuI7pnA/462-2021-performance-bigger-size-sample-for-perf-testing-8

To test,  either run in anger against the performance env and wait 4-5 hrs.   To create a bucket in your own environment you can run the terraform with PERF_SAMPLE_BUCKET=true, you will have to upload the sample file yourself

### CODE REVIEW

File: resources/dummy_keys/1_per_treatment_code.csv [hunk 1]
```diff
@@ -0,0 +1,46 @@
+ARID,ESTAB_ARID,UPRN,ADDRESS_TYPE,ESTAB_TYPE,ADDRESS_LEVEL,ABP_CODE,ORGANISATION_NAME,ADDRESS_LINE1,ADDRESS_LINE2,ADDRESS_LINE3,TOWN_NAME,POSTCODE,LATITUDE,LONGITUDE,OA,LSOA,MSOA,LAD,REGION,HTC_WILLINGNESS,HTC_DIGITAL,TREATMENT_CODE,FIELDCOORDINATOR_ID,FIELDOFFICER_ID,CE_EXPECTED_CAPACITY
```
Comment (NickGrantONS) [inline resources/dummy_keys/1_per_treatment_code.csv]: why is this file stored in the `dummy_keys` directory?

File: features/environment.py [hunk 2]
```diff
@@ -16,7 +16,11 @@ def before_all(_):
     logging.captureWarnings(True)
 
 
-def before_scenario(context, _):
+def before_scenario(context, scenario):
```
Comment (liamtoozer) [inline features/environment.py]: Might be worth mentioning in the README about adding the `@skip` tag to the scenario which tests the 3.5 mil sample if running locally? Either that, or maybe we could skip over this test if running locally by tagging the scenario with `skip-local-dev` and then add that tag to the Makefile to ignore when these tests are run locally?

File: run_gke.sh [hunk 3]
```diff
@@ -53,4 +53,6 @@ kubectl run performance-tests -it --command --rm --quiet --generator=run-pod/v1
     --env=RABBITMQ_PASSWORD=$(kubectl get secret rabbitmq -o=jsonpath="{.data.rabbitmq-password}" | base64 --decode) \
     --env=RABBITMQ_MAN_PORT=15672 \
     --env=SAMPLE_FILE_PATH=resources/sample_files/1_per_treatment_code.csv \
+    --env=SAMPLE_BUCKET=census-rm-performance-test-files \
```
Comment (liamtoozer) [inline run_gke.sh]: This doesn't match the name of the bucket in the terraform PR. Might need to tweak this?

File: config.py [hunk 4]
```diff
@@ -43,6 +43,8 @@ class Config:
     ACTION_RULE_DELAY_MINUTES = float(os.getenv('ACTION_RULE_DELAY_MINUTES', 0.3))
     PRINT_FILE_TIME_LIMIT_MINUTES = float(os.getenv('PRINT_FILE_TIME_LIMIT_MINUTES', 10))
 
-    SAMPLE_FILE_PATH = Path(os.getenv('SAMPLE_FILE_PATH',
-                                      PROJECT_PATH.joinpath('resources', 'sample_files',
-                                                            '100_per_treatment_code.csv')))
+    SAMPLE_FILE_PATH = Path(os.getenv('SAMPLE_FILE_PATH', PROJECT_PATH.joinpath('resources', 'sample_files',
+                                                                                '100_per_treatment_code.csv')))
+
+    SAMPLE_BUCKET = os.getenv('SAMPLE_BUCKET', 'census-rm-performance-test-files')
```
Comment (liamtoozer) [inline config.py]: This doesn't match the name of the bucket in the terraform PR. Might need to tweak this?

File: resources/dummy_keys/1_per_treatment_code.csv (hunk 1 above)
Comment (LukeLoze1) [reply resources/dummy_keys/1_per_treatment_code.csv]: sorted

File: features/environment.py (hunk 2 above)
Comment (LukeLoze1) [reply features/environment.py]: need to decide

**Pull request merged**